Sources/SQLiteNIO/SQLiteConnection.swift

```diff
@@ -3,83 +3,17 @@ import NIOPosix
 import CSQLite
 import Logging
 
-public protocol SQLiteDatabase {
-    var logger: Logger { get }
-    var eventLoop: any EventLoop { get }
-    
-    @preconcurrency func query(
-        _ query: String,
-        _ binds: [SQLiteData],
-        logger: Logger,
-        _ onRow: @escaping @Sendable (SQLiteRow) -> Void
-    ) -> EventLoopFuture<Void>
-    
-    @preconcurrency func withConnection<T>(
-        _: @escaping @Sendable (SQLiteConnection) -> EventLoopFuture<T>
-    ) -> EventLoopFuture<T>
-}
-
-extension SQLiteDatabase {
-    @preconcurrency
-    public func query(
-        _ query: String,
-        _ binds: [SQLiteData] = [],
-        _ onRow: @escaping @Sendable (SQLiteRow) -> Void
-    ) -> EventLoopFuture<Void> {
-        self.query(query, binds, logger: self.logger, onRow)
-    }
-    
-    public func query(
-        _ query: String,
-        _ binds: [SQLiteData] = []
-    ) -> EventLoopFuture<[SQLiteRow]> {
-        let rows: UnsafeMutableTransferBox<[SQLiteRow]> = .init([])
-        return self.query(query, binds, logger: self.logger) { row in
-            rows.wrappedValue.append(row)
-        }.map { rows.wrappedValue }
-    }
-  }
-
-extension SQLiteDatabase {
-    public func logging(to logger: Logger) -> any SQLiteDatabase {
-        _SQLiteDatabaseCustomLogger(database: self, logger: logger)
-    }
-}
-
-private struct _SQLiteDatabaseCustomLogger: SQLiteDatabase {
-    let database: any SQLiteDatabase
-    var eventLoop: any EventLoop {
-        self.database.eventLoop
-    }
-    let logger: Logger
-    
-    @preconcurrency func withConnection<T>(
-        _ closure: @escaping @Sendable (SQLiteConnection) -> EventLoopFuture<T>
-    ) -> EventLoopFuture<T> {
-        self.database.withConnection(closure)
-    }
-    
-    @preconcurrency func query(
-        _ query: String,
-        _ binds: [SQLiteData],
-        logger: Logger,
-        _ onRow: @escaping @Sendable (SQLiteRow) -> Void
-    ) -> EventLoopFuture<Void> {
-        self.database.query(query, binds, logger: logger, onRow)
-    }
-}
-
-internal final class SQLiteConnectionHandle: @unchecked Sendable {
+final class SQLiteConnectionHandle: @unchecked Sendable {
```

Comment:
How are we ensuring that this is safe? IIRC it's bound to an event loop right?

Reply:
Connection handles are safed by libsqlite itself (`SQLITE_THREADSAFE=1` plus `SQLITE_OPEN_FULLMUTEX`), and we further protect it by only ever using it on an `NIOThreadPool`. This is just to shut up the compiler about the use of an `OpaquePointer`.

Reply:
Could we add a comment to that effect?

Reply:
Turns out I already added one in local changes I haven't pushed up yet 😅